benchmark/multithread.cpp

```diff
@@ -0,0 +1,110 @@
+/*
+ *
+ * Copyright (C) 2024 Intel Corporation
+ *
+ * Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ *
+ */
+
+#include "multithread.hpp"
+
+#include <umf/memory_pool.h>
+#include <umf/pools/pool_jemalloc.h>
+#include <umf/pools/pool_scalable.h>
+#include <umf/providers/provider_os_memory.h>
+
+#include <iostream>
+#include <memory>
+#include <numeric>
+
+static constexpr size_t N_REPEATS = 5;
+static constexpr size_t N_ITERATIONS = 50000;
+static constexpr size_t N_THREADS = 20;
+static constexpr size_t ALLOC_SIZE = 64;
+
+using poolCreateExtParams = std::tuple<umf_memory_pool_ops_t *, void *,
+                                       umf_memory_provider_ops_t *, void *>;
+
+static auto poolCreateExtUnique(poolCreateExtParams params) {
```

Comment:
I think we already have this function defined in poolFixtures.hpp - are they the same? If not - please change the name of this and move it to poolFixtures if this is possible

Reply:
But poolFixtures.hpp is an internal header in tests. Do we want to use test headers in benchmarks/examples? This would also mean importing a lot of other stuff (including some gtest headers probably). I think it's better to just have a copy of this function here.

Reply:
ok